runtime/src/App.tsx

```diff
@@ -437,19 +438,26 @@ export default class App extends React.Component<object, State> {
     let rootElement: React.ReactElement | undefined;
     try {
       const rootModuleUri = 'module://' + Files.entry();
-
       if (changedPaths.length > 0) {
         await Modules.flush({ changedPaths, changedUris: [rootModuleUri] });
       }
 
-      const hasRootModuleUri = await Modules.has(rootModuleUri);
-      if (!hasRootModuleUri) {
-        const rootDefaultExport = (await Modules.load(rootModuleUri)).default;
-        if (!rootDefaultExport) {
-          throw new Error(`No default export of '${Files.entry()}' to render!`);
+      // Special handling for Expo Router projects
+      if (Modules.hasDependency('expo-router')) {
+        // NOTE(cedric): this is the default `require.context('./app')` virtual module equivalent, see `snack-require-context`
+        const ctx = await Modules.load('module://app?ctx=eyJyIjp0cnVlLCJtIjoiLioiLCJvIjoic3luYyJ9');
```

Comment:
nix: maybe we could move the `module://app?ctx=eyJyIjp0cnVlLCJtIjoiLioiLCJvIjoic3luYyJ9` as an export constant from `ExpoRouterEntry`. it would be clear if we put all things together.

Reply:
Sounds good